### ISSUE TITLE
will not shutdown fully or restart at all any more, stuck at blank screen with backlight on

### ISSUE THREAD

**Issue by nipsy** [ValveSoftware/SteamOS#1068]
#### Your system information

* Steam client version: 1688001191
* SteamOS version: 3.5
* Opted into Steam client beta?: Yes
* Opted into SteamOS beta?: Yes 
* Have you checked for updates in Settings > System?: Yes

#### Please describe your issue in as much detail as possible:
I had reason to perform a full re-image of my 512GB Steam Deck recently.  I actually started though by performing just the reinstall to avoid wiping all my data.  But after doing the reinstall, my Deck wasn't shutting down fully.  The animated Valve logo screen would appear where it says "Shutting down Steam..." and then the screen would blank, but the backlight would still be visible and I could still hear a fan faintly humming through the vent, so it's clearly wasn't shutting down all the way.  Holding down the power button at this point finally turns it off completely of course, but hard to say without plugging more stuff in what state the system was in exactly at this point.  I thought that perhaps my unrelated screwing around had potentially caused this situation, so I opted for the full re-image option from the recovery image.

After the initial setup was completed, I immediately tested out shutting down the system before doing any other unnecessary updates of the OS or Steam itself, beyond what the initial setup does.  Everything worked as expected at this point with the system fully shutting down as it always has done.  I thought my problem had been fixed!

I went ahead and flipped on the advanced update channels under the developer menu and then opted into the main OS channel and the beta Steam channel and performed the updates.  I had also installed only one game (Brotato) before doing this.  After rebooting from the update, I'm back to a system that doesn't shutdown.  And while typing up this issue, I decided to check whether even a restart would work, and it's doing the exact same thing where it says "Shutting down Steam..." and then getting stuck at a blank screen with the backlight on.

Since I'm now using a bog standard installation without any craziness having happened on the currently installed system otherwise, this seems like a fairly major regression, hence this bug report.  While trying to find information about other people experiencing this same issue over the last few days, I did see something about the storage maintenance task being potentially related.  However, I checked that button in my UI and it doesn't appear that the task is currently running.  So this doesn't seem to be the problem in my situation at least.  I did purchase the Valve dock a few weeks ago expressly to be able to re-image the system as a result of my pacman misadventures that I vaguely referenced earlier and it's the only dock I've ever used with my Deck.  I can't possibly see how it would be related at all, but I thought I'd mention it at least just to cover my bases.  Although if using the dock with some attached peripherals or setting up this version of the system again to allow me to SSH into it might help in troubleshooting anything, let me know and I'll certainly endeavor to provide whatever information could prove useful.

#### Steps for reproducing this issue:

1. Fully re-image Deck
2. Update to Main/Beta
3. Shutdown/Restart (maybe install one small thing?)

**Comment by RodoMa92** [comment]
Can confirm the same thing since the last main update, it seems that game scope loves to lockup when restarting. I also have issues now on waking up, I just get a black screen.

It's a main regression since 6.1.29 kernel update.

**Comment by JDGBOLT** [comment]
I think these are sort of 2 separate issues. The gamescope-session user systemd unit gets stuck when you reboot/shut down the deck. The black screen on waking up the device was introduced in today's Beta Steam Client update, which has steam completely hard lockup, and even if you stop the gamescope-session service, steam gets stuck running with a steamwebhelper process going zombie.

**Comment by JDGBOLT** [comment]
It looks like what is getting stuck on with the gamescope session is the following according to systemctl:
             ├─18340 /usr/bin/ibus-daemon -d -r --panel=disable --emoji-extension=disable
             ├─18342 /usr/bin/python /usr/lib/hwsupport/power-button-handler.py
             └─18354 /usr/lib/ibus/ibus-dconf
If you kill ibus-daemon and power-button-handler.py then steam will restart gracefully, so it seems like something is forcing those processes to still be running when you exit steam.

**Comment by unclejack** [comment]
I've run into the very same issue yesterday with the latest SteamOS 3.5 image from main. I've seen this happen twice on shutdown yesterday after updating the software.

I'm forced to use SteamOS 3.5 on my Steam Deck due to the issues described in #1015.

**Comment by Managor** [comment]
It may or may not be related that after briefly using the Main channel and then going back to the Beta channel, my deck takes abnormally long to boot and wake up. The Main channel might have tainted my install.

**Comment by azridere** [comment]
Experiencing both issues here; Deck stuck at black screen when trying to restart/shutdown, AND failing to wake up from hibernate.
OS Update Channel: Main
Client Update Channel: Steam Deck Beta
I haven't moved from these channels for around 2 months now, so swapping between them isn't the issue for me.

I noticed this issue start around 2 updates ago, having to restart my deck each time I want to play it is a bit annoying, hope a fix can be found soon!

**Comment by JDGBOLT** [comment]
Just for reference the main update from earlier today, 20230630.1000 has fixed the shutdown/reboot freezing issue, seems to behave as it should. Current beta client is still affected and will fail to resume from sleep.

**Comment by azridere** [comment]
20230630.1000 hasn't rolled out to me yet it seems ;P
I'll confirm here once I'm able to update

**Comment by RodoMa92** [comment]
Same for me, there is no 0630, just 0628. Not sure if it got pulled for other unrelated issues. The new steam client update however fixed the wakeup issue of the deck, so at least that's gone.

**Comment by Managor** [comment]
After the latest steam update, the boot times are back to normalcy for me too.

**Comment by eifelkenny** [comment]
I still have the shutdown issue on 3.5. I habe the latest beta 20230628.1000 installed. Is there something I can do?

**Comment by drlamb** [comment]
Thank you for creating this issue. After doing a shell-swap I thought I had inadvertently damaged something causing my deck to not fully shut off.

**Comment by RodoMa92** [comment]
To fix this and update to 0630.1000, you need to downgrade to beta or stable and then update back to main. Valve has messed up some versioning system on 0628 and onwards and for whatever reason on both 0628 and 0630 will still see only 0628 as the latest version. When you install 0630 it still will claim a new OS update for 0628, do not install that for now, wait for next wekk when they will probably rectify it. At least on 30 reboot works perfectly fine again.

**Comment by unclejack** [comment]
Can we get a proper image which doesn't require workarounds, please? I'd much rather not go through additional steps to switch to the 0630 image. I'm already running the main image to work avoid instability on the 3.4 branch of SteamOS.

While I wholeheartedly appreciate what Valve is doing, it's also odd to see some of the bugs which are unique to SteamOS and to the Steam Deck.

**Comment by Managor** [comment]
Patience. This is the Main channel we're talking about. It's even hidden behind an advanced option. They'll get to it when they get to it.

**Comment by unclejack** [comment]
> Patience. This is the Main channel we're talking about. It's even hidden behind an advanced option. They'll get to it when they get to it.

This was meant for those who work at Valve or for Valve. I've been rather patient for a while now. It's not just this one issue. Regardless, have a good day.

**Comment by RodoMa92** [comment]
It's fixed for me as of now. The latest version is now correctly 0630.

**Comment by nipsy** [comment]
Yep, working on my end for sure as of at least OS 20230630.1000 and Steam 1688792985.  Closing this report.

**Comment by unclejack** [comment]
That new image was also downloaded and installed on my Steam Deck a few days ago.

The one thing to be aware of is that it might freeze on a black screen when updating from a previous image. It did that in my case. I left the Steam Deck as it was for 5 minutes to let it finish any potentially unfinished disk writes and turned it off. It booted using the updated SteamOS image.

The new SteamOS image fixes the hang on shutdown. The issue can be closed.

**Comment by kudu9** [comment]
I have something samullare now, about a month ago I reimaged my steam deck and just now I was playing days gone and the game start runing at 6 FPS and everything was laggy af, so I closed the game but the whole steam deck was lagging so I did a restart and the steam deck just boot, it showes the steam deck logo and then just blank screen while the backlight is on :( I really don't wanna re image (I can access the bios just fine)

**Comment by unclejack** [comment]
@kudu9: The issue you've run into is a different one. The bug reported initially has been fixed.

You probably want to contact Steam support if you haven't done so already.
